src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,855 @@
+            WaitForADC();
+        }
+
+        /// <summary>
+        /// Waits after a single conversion until the DOUT/!RDY goes low to indicate the completion of a conversion
+        /// </summary>
+        public void WaitForADC()
+        {
+            while (!IsReady)
+            {
+                Thread.Sleep(5);
+            }
+        }
+
+        /// <summary>
+        /// Initiates Single Conversion (device will go into low power mode when conversion complete, and DOUT/!RDY goes low to indicate the completion of a conversion)
+        /// </summary>
+        public void StartSingleConversion()
+        {
+            _registerCache[(byte)Register.Mode] = SetRegisterBits(Register.Mode, BitMask.ModeMD, 0b001);      // single conversion mode bits
+            SetRegisterValue(Register.Mode, _registerCache[(byte)Register.Mode]);
+
+            _stopWatch.Restart();
+        }
+
+        /// <summary>
+        /// Initiates Continuous Conversion. After calling this method the AdcValueReceived event will be fired continuously. This mode is much faster, but uses significatily more CPU power.
+        /// </summary>
+        /// <param name="frequency">The target frequency of the sampling. AD7193 has the maximum of 4800 samples per second.</param>
+        public void StartContinuousConversion(uint frequency = 0)
+        {
+            if (frequency == 0)
+            {
+                frequency = metadata.ADCSamplerate;
+            }
+
+            if (frequency > metadata.ADCSamplerate)
+            {
+                throw new ArgumentException($"The frequency you provided is higher than the maximum sampling rate of AD7193 ({metadata.ADCSamplerate} SPS).");
+            }
+
+            _registerCache[(byte)Register.Mode] = SetRegisterBits(Register.Mode, BitMask.ModeMD, 0b000);      // continuous conversion mode
+            SetRegisterValue(Register.Mode, _registerCache[(byte)Register.Mode]);
+
+            ContinuousRead = true;
+
+            long samplePerTicks = Stopwatch.Frequency / frequency;
+            if (samplePerTicks == 0)
+            {
+                samplePerTicks = 1;
+            }
+
+            _stopWatch.Restart();
+            Task samplingTask = Task.Run(() =>
+            {
+                long samples = 0;
+                long nextSampleAt = _stopWatch.ElapsedTicks;
+                long elapsedTicks = 0;
+                long jitter = 0;
+                long maxJitterCorrectionPerSample = Math.Max(Stopwatch.Frequency / 100000, 1);
+                while (_stopWatch.IsRunning)
+                {
+                    elapsedTicks = _stopWatch.ElapsedTicks;
+                    if (elapsedTicks >= nextSampleAt)
+                    {
+                        nextSampleAt = elapsedTicks + samplePerTicks;
+                        ReadADCValue();
+                        samples++;
+                        jitter = (elapsedTicks - (samples * samplePerTicks));
+                        if (JitterCorrection)
```

Comment:
i have a mixed feelings on adding utilities like that to devices. One one hand they're useful but on the other hand I suspect these might be useful elsewhere as well. Can you please write in the docs of this device more about this so that if I find similar code to be useful in the future for different device we could unify those?

Reply:
Yes, I had mixed feelings as well. It would be great to have a base class for all the ADCs, so the helper methods like these could go there, unifying the way they do the processing.
At this point we don't have a device class hierarchy, so I think it is a good place to have it, for now.